The sanitizer detected a heap buffer overflow vulnerability. The explanation of the vulnerability is: The program tried to access a heap object outside of its allocated memory, causing potential memory corruption or crashes. Here is the detail: 

[1m[0m[1m[34mWRITE of size 1 at 0x529000018bc0 thread T0[1m[0m
    - tok_get_normal_mode /src/cpython3/Parser/tokenizer.c:2213:37
    - tok_get /src/cpython3/Parser/tokenizer.c:0:0
    - _PyTokenizer_Get /src/cpython3/Parser/tokenizer.c:2645:18
    - _PyPegen_tokenize_full_source_to_check_for_errors /src/cpython3/Parser/pegen_errors.c:170:17
    - _PyPegen_run_parser /src/cpython3/Parser/pegen.c:844:9
    - _PyPegen_run_parser_from_string /src/cpython3/Parser/pegen.c:938:14
    - Py_CompileStringObject /src/cpython3/Python/pythonrun.c:1771:11
    - builtin_compile_impl /src/cpython3/Python/bltinmodule.c:831:14
    - builtin_compile /src/cpython3/Python/clinic/bltinmodule.c.h:383:20
    - cfunction_vectorcall_FASTCALL_KEYWORDS /src/cpython3/Objects/methodobject.c:438:24
    - _PyObject_VectorcallTstate /src/cpython3/Include/internal/pycore_call.h:92:11
    - _PyEval_EvalFrameDefault /src/cpython3/Python/bytecodes.c:2533:19
    - _PyObject_VectorcallTstate /src/cpython3/Include/internal/pycore_call.h:92:11
    - PyObject_CallOneArg /src/cpython3/Objects/call.c:379:12
    - fuzz_ast_literal_eval /src/cpython3/Modules/_xxtestfuzz/fuzzer.c:425:25
    - _run_fuzz /src/cpython3/Modules/_xxtestfuzz/fuzzer.c:443:14
    - LLVMFuzzerTestOneInput /src/cpython3/Modules/_xxtestfuzz/fuzzer.c:568:11
    - fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:614:13
    - fuzzer::RunOneTest(fuzzer::Fuzzer*, char const*, unsigned long) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerDriver.cpp:327:6
    - fuzzer::FuzzerDriver(int*, char***, int (*)(unsigned char const*, unsigned long)) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerDriver.cpp:862:9
    - main /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerMain.cpp:20:10

[1m[32m0x529000018bc0 is located 16 bytes after 18864-byte region [0x529000014200,0x529000018bb0)
[1m[0m[1m[35mallocated by thread T0 here:[1m[0m
    - malloc /src/llvm-project/compiler-rt/lib/asan/asan_malloc_linux.cpp:68:3
    - tok_new /src/cpython3/Parser/tokenizer.c:83:49
    - _PyTokenizer_FromUTF8 /src/cpython3/Parser/tokenizer.c:880:29
    - _PyPegen_run_parser_from_string /src/cpython3/Parser/pegen.c:913:15
    - Py_CompileStringObject /src/cpython3/Python/pythonrun.c:1771:11
    - builtin_compile_impl /src/cpython3/Python/bltinmodule.c:831:14
    - builtin_compile /src/cpython3/Python/clinic/bltinmodule.c.h:383:20
    - cfunction_vectorcall_FASTCALL_KEYWORDS /src/cpython3/Objects/methodobject.c:438:24
    - _PyObject_VectorcallTstate /src/cpython3/Include/internal/pycore_call.h:92:11
    - _PyEval_EvalFrameDefault /src/cpython3/Python/bytecodes.c:2533:19
    - _PyObject_VectorcallTstate /src/cpython3/Include/internal/pycore_call.h:92:11
    - PyObject_CallOneArg /src/cpython3/Objects/call.c:379:12
    - fuzz_ast_literal_eval /src/cpython3/Modules/_xxtestfuzz/fuzzer.c:425:25
    - _run_fuzz /src/cpython3/Modules/_xxtestfuzz/fuzzer.c:443:14
    - LLVMFuzzerTestOneInput /src/cpython3/Modules/_xxtestfuzz/fuzzer.c:568:11
    - fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:614:13
    - fuzzer::RunOneTest(fuzzer::Fuzzer*, char const*, unsigned long) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerDriver.cpp:327:6
    - fuzzer::FuzzerDriver(int*, char***, int (*)(unsigned char const*, unsigned long)) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerDriver.cpp:862:9
    - main /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerMain.cpp:20:10


To fix this issue, follow the advice below:

1. If overflow is unavoidable, allocate a sufficiently large buffer during initialization.
2. Add explicit bounds checking before accessing arrays or buffers to prevent overflows.
3. Replace unsafe functions like memcpy, strcpy, strcat, and sprintf with safer alternatives such as strncpy, strncat, and snprintf.
4. Check for integer overflows in size calculations that could cause improper memory allocations.
